worldedit-core/src/main/java/com/fastasyncworldedit/core/queue/implementation/chunk/ChunkHolder.java

```diff
@@ -992,9 +1038,19 @@ public synchronized <V extends IChunk> void init(IQueueExtent<V> extent, int chu
 
     @Override
     public synchronized T call() {
-        if (chunkSet != null) {
+        calledLock.lock();
+        if (chunkSet != null && !chunkSet.isEmpty()) {
             chunkSet.setBitMask(bitMask);
-            return this.call(chunkSet, this::recycle);
+            try {
+                return this.call(chunkSet, () -> {
+                    recycle();
+                    calledLock.unlock();
+                });
+            } catch (Throwable t) {
+                calledLock.unlock();
+                t.printStackTrace();
```

Comment:
If the exception is rethrown, we shouldn't print it. 
How can we make sure that the callback/finalizer is actually called (on the same thread) to actually release the lock?